### DISCUSSION TITLE
Proxy agent granular configuration

### DISCUSSION THREAD

**Discussion by eugeniosegala** [panva/openid-client#630] [Q&A] [answered]
Hello!

I'm configuring `openid-client` to use a proxy and I read the documentation regarding the [customisations](https://github.com/panva/node-openid-client/blob/main/docs/README.md#customizing-http-requests).

I wanted to confirm that this is the correct way to setup a proxy across all layers (Issuer discover, refresh operations etc...).

```javascript

import { Issuer, custom } from "openid-client";

const issuerUrl = `issuer-url`;

Issuer[custom.http_options] = function () {
  return {
    agent: proxyAgent,
  };
};

const keycloakIssuer = await Issuer.discover(issuerUrl);

keycloakIssuer[custom.http_options] = function () {
  return {
    agent: proxyAgent,
  };
};

keycloakClient = new keycloakIssuer.Client({
  // my config...
});

keycloakClient[custom.http_options] = function () {
  return {
    agent: proxyAgent,
  };
};
```

Ignore the code repetition for now as I just wanted to make clear that I needed to configure the proxy 3 times!

Am I getting this right?

Thanks!

**Comment by panva** [comment, marked as answer]
Yes, your approach works. 

I believe this will do as well if you only have this one issuer in your codebase.

```
import { custom } from 'openid-client';

custom.setHttpOptionsDefaults({
  agent: proxyAgent,
});
```
